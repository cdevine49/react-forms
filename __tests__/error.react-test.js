import React from 'react';
import Error from '../src/components/error';
import renderer from 'react-test-renderer';

describe("Errors", () => {
  test("does not render without child", () => {
    const tree = renderer.create(
      <Error
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders with child", () => {
    const tree = renderer.create(
      <Error>Test Error</Error>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("accepts props", () => {
    const tree = renderer.create(
      <Error
        id="errorId"
        className="error-class test"
        >
        Test Error
      </Error>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});