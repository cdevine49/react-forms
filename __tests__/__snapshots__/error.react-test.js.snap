// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors accepts props 1`] = `
<p
  className="error-message error-class test"
  id="errorId"
>
  Test Error
</p>
`;

exports[`Errors does not render without child 1`] = `null`;

exports[`Errors renders with child 1`] = `
<p
  className="error-message"
>
  Test Error
</p>
`;
